.github/workflows/build.yaml

```diff
@@ -82,3 +93,41 @@ jobs:
       with:
         name: Widelands ${{ matrix.config }} ${{ matrix.arch }} Installer
         path: ${{ github.workspace }}\Widelands-${{ github.sha }}-${{ matrix.config }}-${{ matrix.arch }}.exe
+
+  macos:
+    needs: [clang_tidy]
+    strategy:
+      matrix:
+        config:
+          - release
+          - debug
+        compiler:
+          - clang
+    name: MacOs ${{ matrix.config }} ${{ matrix.compiler }}
+    runs-on: macOS-latest
+    steps:
+    - name: Installing dependencies
+      run: |
+        export ICU_ROOT="$(brew --prefix icu4c)" && \
+        export PATH="/usr/local/opt/ccache/libexec:/usr/local/opt/gettext/bin:$PATH";
+        brew install boost ccache glew ninja sdl2 sdl2_image sdl2_mixer sdl2_ttf
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 1
+    - name: Building
+      run: |
+        mkdir build_wl
+        cd build_wl/
+        ../utils/macos/build_app.sh ${{ matrix.compiler }} ${{ matrix.config }} ../
+# Code for uploading artifact starts here. It's not being found although the paths match.
+#        DMGPATH="$(pwd)"
+#        DMGFILE="$(ls *.dmg)"
+#        echo "DMG file is: $DMGPATH/$DMGFILE"
+#        echo "::set-env name=dmg::$DMGPATH/$DMGFILE"
+
+#    - name: Uploading DMG
+#      uses: actions/upload-artifact@v2
+#      with:
+#        name: Widelands ${{ matrix.config }} ${{ matrix.compiler }} AppImage
+#        path: ${{ env.dmg }}
```

Comment:
The `env` variables are not being evaluated correctly in this context. Did you try moving the DMG to a hardcoded path (e.g. `${{ github.workspace }}/Widelands-${{ github.sha }}-${{ matrix.config }}.dmg`) and passing that as the artifact `path`?

Reply:
I'd need to reprogram `buildapp.sh` for that, which I don't want to touch because I don't have a Mac. Also, the contents of `${{ env.dmg }}` are correct, so I don't know if that would help.

For example, in this run: https://github.com/gunchleoc/widelands/actions/runs/208860783

It complains about

> No files were found with the provided path: /Users/runner/work/widelands/widelands/build_wl/widelands_10.9_1.0~git1[302d788@build-macos].dmg. No artifacts will be uploaded.

The path is identical to the one created by the build step:

```
Creating DMG ...
created: /Users/runner/work/widelands/widelands/build_wl/widelands_10.9_1.0~git1[302d788@build-macos].dmg
DMG file is: /Users/runner/work/widelands/widelands/build_wl/widelands_10.9_1.0~git1[302d788@build-macos].dmg
```

Reply:
Just do `mv OLD_DANGEROUS_PATHNAME NEW_WELLDEFINED_PATHNAME` directly after buildapp.sh – unless renaming would damage the DMG? The new pathname needs to be passed to the uploader action without accessing any environment variables

Reply:
Thanks, I have pushed an attempt